http-clients/src/main/java/com/palantir/remoting/http/FeignClients.java

```diff
@@ -41,8 +43,11 @@ private FeignClients() {}
     public static FeignClientFactory standard() {
         return FeignClientFactory.of(
                 new GuavaOptionalAwareContract(new JAXRSContract()),
-                new JacksonEncoder(ObjectMappers.guavaJdk7()),
-                new OptionalAwareDecoder(new TextDelegateDecoder(new JacksonDecoder(ObjectMappers.guavaJdk7()))),
+                new InputStreamDelegateEncoder(new JacksonEncoder(ObjectMappers.guavaJdk7())),
+                new OptionalAwareDecoder(
```

Comment:
yeah, something like that. i had pitched essentially the same design to the feign folks a while ago, but it was too flexible for their liking: https://github.com/Netflix/feign/pull/287/files

haven't looked at the OptionalAwareDecoder/Contract interaction in detail, but yeah, remove away :)